internal/pkg/api/server/templates/service/service.go

```diff
@@ -236,8 +237,30 @@ func (s *service) InstanceIndex(d dependencies.Container, payload *InstanceIndex
 	return InstanceResponse(d, prjState, branchKey, payload.InstanceID)
 }
 
-func (s *service) UpdateInstance(dependencies.Container, *UpdateInstancePayload) (res *InstanceDetail, err error) {
-	return nil, NotImplementedError{}
+func (s *service) UpdateInstance(d dependencies.Container, payload *UpdateInstancePayload) (res *InstanceDetail, err error) {
+	prjState, branchKey, instance, err := getTemplateInstance(d, payload.Branch, payload.InstanceID, true)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := renameInst.Options{
+		Branch:   branchKey,
+		Instance: *instance,
+		NewName:  payload.Name,
+	}
+
+	err = renameInst.Run(prjState, opts, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// Push changes
+	changeDesc := fmt.Sprintf("Rename template instance %s", payload.InstanceID)
+	if err := push.Run(prjState, push.Options{ChangeDescription: changeDesc, AllowRemoteDelete: true, DryRun: false, SkipValidation: true}, d); err != nil {
+		return nil, err
+	}
+
+	return InstanceResponse(d, prjState, branchKey, payload.InstanceID)
```

Comment:
@webrouse Problém je v tom, že `InstanceReponse()` bere metadata z `branch.Remote` (https://github.com/keboola/keboola-as-code/blob/6db321ce6d48d7ed04fa31ed44a933c785742555/internal/pkg/api/server/templates/service/mapper.go#L269) nicméně tam to přejmenovaný není, je to jen v `branch.Local`. Tak teď nevím co je nejlepší cesta. Nejčistší by bylo asi zavolat znovu `getTemplateInstance()`, ale vzhledem k těm hackům co čekají na ten refactoring se bojím jestli by to netrvalo nějak extra dlouho. Co myslíš?

Reply:
Pozriem na to za chvilku, mal som neplanovany call s UI :/ a idem si dat este obed

Reply:
Rychly napad, co ta zmenit `Remote` na `LocalOrRemote`? Je tam tusi taka metoda ....

Reply:
@JakubMatejka ^^^ to by mohlo fungovat, skus to vymeniť na tom riadku `269` -> `branch.LocalOrRemoteState().(*model.Branch)`, tj. aby ten `local` state mal prednost, .... ako je to uz hack na hack, ale nic lepsie mi zatial nenapadlo.

Reply:
Jo, jde to. 👍 Díky.